security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Stage 1.4: Kona and Asterisc FMA (Failure Modes and Recovery Path Analysis)](#stage-14-kona-and-asterisc-fma-failure-modes-and-recovery-path-analysis)
+  - [Introduction](#introduction)
+  - [Failure Modes and Recovery Paths](#failure-modes-and-recovery-paths)
+    - [Kona and Asterisc Hot Path Failure](#kona-and-asterisc-hot-path-failure)
+    - [Divergent Kona Derivation Pipeline](#divergent-kona-derivation-pipeline)
+    - [Kona Execution Diverges from op-program](#kona-execution-diverges-from-op-program)
+    - [Breaking Backwards Compatibility for the op-challenger](#breaking-backwards-compatibility-for-the-op-challenger)
+    - [Divergence between RISCV.sol and Asterisc](#divergence-between-riscvsol-and-asterisc)
+    - [Generic items we need to take into account:](#generic-items-we-need-to-take-into-account)
+  - [Audit Requirements](#audit-requirements)
+  - [Appendix](#appendix)
+    - [Appendix A: op-challenger](#appendix-a-op-challenger)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Stage 1.4: Kona and Asterisc FMA (Failure Modes and Recovery Path Analysis)
+
+| | |
+|--------|--------------|
+| Author | Andreas Bigger |
+| Created at | 2024-10-17 |
+| Initial Reviewers | Matt Solomon, Ethnical |
+| Need Approval From | Ben Clabby, Mark Tyneway |
+| Status | In Review |
+
+
+## Introduction
+
+This document covers the initial deployment of [the Asterisc Fault Proof VM][asterisc] with [the Kona Fault Proof Program][kona].
+
+Together, Kona and Asterisc form an alternative proof stack to the [op-program] and [cannon].
+A secondary dispute game will be created that uses kona and asterisc.
+The [op-challenger] supports playing _both_ game types - `op-program` + `cannon` as well as `kona` + `asterisc`.
```

Comment:
For my own understanding: Is `op-program` coupled to `cannon` and `kona` coupled to `asterisc`, or can they be mixed, e.g. run `op-program with `asteristc`? If coupled, what makes them coupled and do we plan to decouple at some point, or do they have to be coupled for some reason?

Reply:
Right now, they are coupled. There's not much that enforces this - we could run `op-program` on `asterisc` and `kona` on `cannon`. However, the two languages require much different things. `kona` needs a much less fully-featured VM than `op-program`, and so we'll likely continue in the direction of keeping the VMs coupled.

This may be revisited, but for now, we should treat them as coupled.